File: src/main/java/no/nav/pto_schema/kafka/json/topic/onprem/EndringPaManuellStatusV1.java
package no.nav.pto_schema.kafka.json.topic.onprem;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.experimental.Accessors;

@Data
@AllArgsConstructor
@NoArgsConstructor
@Accessors(chain = true)
public class EndringPaManuellStatusV1 {
    private String aktorId;
    private boolean erManuell;
}

File: src/main/kotlin/no/nav/pto_schema/kafka/json/topic/onprem/EndringPaManuellStatusV1.kt
package no.nav.pto_schema.kafka.json.topic.onprem

data class EndringPaManuellStatusV1(
    val aktorId: String,
    val erManuell: Boolean,
) {
    companion object {
        @JvmStatic
        fun builder(): Builder = Builder()
    }

    class Builder {
        private var aktorId: String? = null
        private var erManuell: Boolean = false

        fun aktorId(aktorId: String?): Builder = apply { this.aktorId = aktorId }
        fun erManuell(erManuell: Boolean): Builder = apply { this.erManuell = erManuell }

        fun build(): EndringPaManuellStatusV1 = EndringPaManuellStatusV1(aktorId!!, erManuell)
    }
}

